### PR TITLE
Add Maps Activity

### PR DESCRIPTION
**Pull request by rrgodhorus**
Enabled maps in navigation drawer and added Maps Activity
solves issue #8

### CODE REVIEW

File: app/src/main/java/com/macbitsgoa/events/maps/MapsActivity.java [hunk 1]
```diff
@@ -0,0 +1,161 @@
...
+ * Google Maps Activity showing Event Markers.
+ *
+ * @author Rajath Reghunath [rrgodhorus@gmail.com]
+ */
+
+@SuppressWarnings("DuplicateStringLiteralInspection")
+public class MapsActivity extends FragmentActivity implements OnMapReadyCallback {
+
+    private static final int MAP_INIT_CAP = 15;  //initial capacity for number of venues
+    private static final String MAP_CENTRAL_LOC = "/locations/central";
```
Comment (Rushi98) [inline app/src/main/java/com/macbitsgoa/events/maps/MapsActivity.java]: Are these firebase keys? If so, put all firebase keys in separate `public final class FirebaseKeys` class.

File: app/src/main/java/com/macbitsgoa/events/maps/MapsActivity.java [hunk 2]
```diff
@@ -0,0 +1,161 @@
...
+    private static final Venue MAP_DEFAULT_CENTRAL = new Venue("15.392199","73.878609","BITS GOA");
+    private static final String FIREBASE_ERROR_TAG = "MapsActivity";
+    private static final String FIREBASE_ERROR_STRING = "Firebase data retrieval failed";
+    private static final String FIREBASE_STRING_LATITUDE = "latitude";
+    private static final String FIREBASE_STRING_LONGITUDE = "longitude";
+    private static final String FIREBASE_STRING_VENUE_NAME = "venueName";
+
+    private float zoomLevel = MAP_DEFAULT_ZOOM;
+    @SuppressWarnings("TypeMayBeWeakened")
+    private final ArrayList<Venue> venues  = new ArrayList<>(MAP_INIT_CAP);
```
Comment (Rushi98) [inline app/src/main/java/com/macbitsgoa/events/maps/MapsActivity.java]: weaken the type as suggested

File: app/src/main/java/com/macbitsgoa/events/maps/MapsActivity.java [hunk 3]
```diff
@@ -0,0 +1,161 @@
...
+
+@SuppressWarnings("DuplicateStringLiteralInspection")
+public class MapsActivity extends FragmentActivity implements OnMapReadyCallback {
+
+    private static final int MAP_INIT_CAP = 15;  //initial capacity for number of venues
+    private static final String MAP_CENTRAL_LOC = "/locations/central";
+    private static final String MAP_ZOOM_SETTINGS = "/locations/zoom";
+    private static final String MAP_MARKERS_LOC = "/locations/locations";
+    private static final float MAP_DEFAULT_ZOOM = 16.0f;
+    private static final Venue MAP_DEFAULT_CENTRAL = new Venue("15.392199","73.878609","BITS GOA");
```
Comment (Rushi98) [inline app/src/main/java/com/macbitsgoa/events/maps/MapsActivity.java]: you can remove `MAP_` prefix for private variables.

File: app/src/main/java/com/macbitsgoa/events/maps/MapsActivity.java [hunk 4]
```diff
@@ -0,0 +1,161 @@
...
+    private static final String FIREBASE_ERROR_TAG = "MapsActivity";
```
Comment (Rushi98) [inline app/src/main/java/com/macbitsgoa/events/maps/MapsActivity.java]: use `TAG_PREFIX + ClassName.class.getSimpleName()`

File: app/src/main/java/com/macbitsgoa/events/maps/MapsActivity.java [hunk 5]
```diff
@@ -0,0 +1,161 @@
...
+            mapFragment.getMapAsync(this);
+        }
+    }
+
+
+    /**
+     * Manipulates the map once available.
+     * This callback is triggered when the map is ready to be used.
+     * This is where we can add markers or lines, add listeners or move the camera. In this case,
+     * we just add a marker near Sydney, Australia.
```
Comment (Rushi98) [inline app/src/main/java/com/macbitsgoa/events/maps/MapsActivity.java]: delete this line

File: app/src/main/java/com/macbitsgoa/events/maps/MapsActivity.java [hunk 6]
```diff
@@ -0,0 +1,161 @@
...
+            public void onDataChange(@NonNull final DataSnapshot dataSnapshot) {
+                for (final DataSnapshot snapshot : dataSnapshot.getChildren()) {
+                    zoomLevel = Float.parseFloat(snapshot.getValue(String.class));
+
+                }
+            }
+
+            @Override
+            public void onCancelled(@NonNull final DatabaseError databaseError) {
+                Log.e(FIREBASE_ERROR_TAG,FIREBASE_ERROR_STRING + " in zoom");
```
Comment (Rushi98) [inline app/src/main/java/com/macbitsgoa/events/maps/MapsActivity.java]: `TAG, databaseError.getMessage, databaseError.toException`

File: app/src/main/java/com/macbitsgoa/events/maps/MapsActivity.java (hunk 5 above)
Comment (rrgodhorus) [reply app/src/main/java/com/macbitsgoa/events/maps/MapsActivity.java]: The whole comment or just the sydney part?

Comment (Rushi98) [reply app/src/main/java/com/macbitsgoa/events/maps/MapsActivity.java]: sydney part

**Comment by rrgodhorus** [comment]
Requested changes implemented. Please review

File: app/src/main/java/com/macbitsgoa/events/maps/Venue.java [hunk 7]
```diff
@@ -0,0 +1,33 @@
+package com.macbitsgoa.events.maps;
+
+class Venue {
+
+    private final String latitude;
```
Comment (Rushi98) [inline app/src/main/java/com/macbitsgoa/events/maps/Venue.java]: make all model attributes `public`
make lat and long `float`
remove getters and setters

File: app/src/main/java/com/macbitsgoa/events/maps/MapsActivity.java [hunk 8]
```diff
@@ -0,0 +1,158 @@
...
+ *
+ * @author Rajath Reghunath [rrgodhorus@gmail.com]
+ */
+
+@SuppressWarnings("DuplicateStringLiteralInspection")
+public class MapsActivity extends FragmentActivity implements OnMapReadyCallback {
+
+    private static final int INIT_CAP = 15;  //initial capacity for number of venues
+    private static final float DEFAULT_ZOOM = 16.0f;
+    private static final Venue DEFAULT_CENTRAL = new Venue("15.392199","73.878609","BITS GOA");
```
Comment (Rushi98) [inline app/src/main/java/com/macbitsgoa/events/maps/MapsActivity.java]: get these values from build config
```gradle
buildConfigField "float", "mapDefCentralLat", "15.392199f"
buildConfigField "float", "mapDefCentralLng", "73.878609"
buildConfigField "String", "mapDefCentralTag", "BITS Goa"
```

**Comment by rrgodhorus** [comment]
Implemented changes.Please review.

**Pull request merged**